packages/hawtio/src/ui/page/HawtioPage.tsx

```diff
@@ -17,22 +17,26 @@ import { PageContext } from './context'
 import { log } from './globals'
 
 export const HawtioPage: React.FunctionComponent = () => {
-  const { username, isLogin, userLoaded } = useUser()
+  const { username, isLogin, userLoaded, userLoading } = useUser()
   const { plugins, pluginsLoaded } = usePlugins()
   const navigate = useNavigate()
   const { search } = useLocation()
   const { selectedNode, setSelectedNode } = usePluginNodeSelected()
 
-  if (!userLoaded || !pluginsLoaded) {
+  // navigate should be used in effect
+  // otherwise "Cannot update a component (`BrowserRouter`) while rendering a different component" is thrown
+  useEffect(() => {
+    if (!isLogin && !userLoading) {
+      navigate('login')
+    }
+  }, [isLogin, navigate, userLoading])
+
+  if (!userLoaded || !pluginsLoaded || userLoading) {
```

Comment:
I still don't get why we start to need this change. We've been doing this without that error and `useEffect()`. What changes made it to be needed?

Reply:
I was getting React error like this: "state modified when running component function".
`navigate('login')` is in fact updating state (react router) and I found that state should be updated in event handlers or effects.
But of course I may be wrong - I'm not (yet?) expert on async React... I'll monitor this fragment - anyway, you know where to find me ;)